include/net/mqtt_socket.h

```diff
@@ -0,0 +1,655 @@
+/*
```

Comment:
The commit subject "net: mqtt: Add MQTT socket implementation" contains redundant information, what about just saying "net: mqtt: Add BSD socket implementation"

Reply:
Good point, will change.